src/OpenSage.Game/Logic/Object/GameObject.cs

```diff
@@ -1397,6 +1521,59 @@ private void CheckDisabledStates()
         }
     }
 
+    public bool IsDisabledByType(DisabledType type) => _disabledTypes.Get(type);
+
+    public void SetDisabled(DisabledType type)
+    {
+        // TODO(Port): Should be FOREVER. See Object::setDisabled()
+        Disable(type, new LogicFrame(0x3fffffff));
+    }
+
+    public void ClearDisabled(DisabledType type)
+    {
+        // TODO(Port): Implement this properly. See Object::clearDisabled()
+        UnDisable(type);
+    }
+
+    public void SetCaptured(bool isCaptured)
+    {
+        if (!isCaptured)
+        {
+            throw new ArgumentException("Clearing captured status. This should never happen.");
```

Comment:
Is this true? What if a tech building is captured by a player, and then that player dies? Don't the objects revert to neutral?

(I'm assuming this behavior was ported, so I'm just curious)

Reply:
Yes, this was ported from [here](https://github.com/electronicarts/CnC_Generals_Zero_Hour/blob/0a05454d8574207440a5fb15241b98ad0b435590/GeneralsMD/Code/GameEngine/Source/GameLogic/Object/Object.cpp#L2001). I think there might be a few definitions of "captured", and this one here is specifically about the "private status" flag.